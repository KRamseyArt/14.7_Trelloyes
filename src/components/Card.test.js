import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

describe('Card component', () =>{
  it('renders without crashing', () =>{
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () =>{
    const tree = renderer
      .create(<Card title="Card" content="something"/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the UI without title and content', () =>{
    const tree = renderer
      .create(<Card />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})