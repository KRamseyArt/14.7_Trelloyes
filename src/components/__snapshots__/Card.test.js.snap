// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3>
    Card
  </h3>
  <p>
    something
  </p>
</div>
`;

exports[`Card component renders the UI without title and content 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    Delete
  </button>
  <h3 />
  <p />
</div>
`;
